test/fast/Unit tests/nvm_download

```diff
@@ -15,4 +16,12 @@ nvm_download "https://raw.githubusercontent.com/nvm-sh/nvm/HEAD/install.sh" >/de
 # nvm_download should fail to download wrong_install.sh
 ! nvm_download "https://raw.githubusercontent.com/nvm-sh/nvm/HEAD/wrong_install.sh" >/dev/null || die "nvm_download should fail to download no existing file"
 
+# nvm_download should pass when calling with auth header
+docker pull kennethreitz/httpbin && docker run -d --name httpbin -p 80:80 kennethreitz/httpbin
+sleep 1 # wait for httpbin to start
+NVM_AUTH_HEADER="Bearer test-token"
+nvm_download "http://127.0.0.1/bearer" > /dev/null || die "nvm_download unable to send auth header"
+unset NVM_AUTH_HEADER
```

Comment:
let's also do a test _without_ the auth header, to demonstrate that the change works - iow, with these two tests, and without your change, the tests should fail